benchmark/nixlbench/src/worker/nixl/nixl_worker.cpp

```diff
@@ -687,10 +875,37 @@ xferBenchNixlWorker::allocateMemory(int num_threads) {
             }
         }
 
+        // Add signal buffer for GDAKI if enabled
+        if (is_gdaki_enabled) {
+#ifdef HAVE_NIXL_DEV_API
+            size_t sigsize;
+            nixl_opt_args_t extra_params = {.backends = {backend_engine}};
+
+            CHECK_NIXL_ERROR(agent->getGpuSignalSize(sigsize, &extra_params),
+                             "getGpuSignalSize failed");
+
+            std::optional<xferBenchIOV> signal_desc = initSignalBuffer(0, sigsize);
+            if (signal_desc) {
+                iov_list.push_back(signal_desc.value());
+                signal_buffers.push_back(signal_desc.value());
```

Comment:
can it be done in some specific device api method instead of general allocation one?